keps/sig-network/2634-kubeproxy-staging/README.md

```diff
@@ -0,0 +1,187 @@
+the evolution of proxy features without being tighten to kubernetes main code and also 
+to future kube-proxy implementations to import this logic without needing to import the 
+whole Kubernetes code.
+
+This was also motivated by [kubernetes #92369](https://github.com/kubernetes/kubernetes/issues/92369)
+
+### Goals
+
+* Move kube-proxy repo to `kubernetes/kube-proxy` repo using staging.
+* Decouple kube-proxy from internal kubernetes dependencies.
+
+### Non-Goals
+* Implementing interfaces for kube-proxy new implementations (like virtual-kubelet, 
+but for proxies). This can be addressed on a later KEP
+
+## Proposal
+
+kube-proxy already have its own repo. The proposal is to:
+
+### Move the internal network dependencies that kube-proxy already has
+kube-proxy have some internal dependencies, mostly notable in `kubernetes/pkg/utils`. We
+can nominate some of them being `iptables`, `ipvs` and `conntrack` packages, but there are
+others as well.
+
+We need to move these dependencies to some repo that does not force kube-proxy to also
+vendor the whole kubernetes repo.
+
+<<[UNRESOLVED repository for network utils ]>>
+Although the majority of utils used by kube-proxy can be moved to component-helpers,
+some still vendors external dependencies (like `moby/ipvs`).
+If we move every pkg/utils used by kube-proxy to component helpers, another project
+already vendoring it (like kubectl) might vendor moby/ipvs because of transitive 
+dependencies.
+We need to define if network utils should be moved to its own repo (suggesting kubernetes/net-utils)
+or if we can accept that other projects vendoring `component-helpers` are going to vendor the 
+dependencies of kube-proxy as well
+<<[/UNRESOLVED]>>
+
+### Move other dependencies kube-proxy already has inside k/k
+kube-proxy vendors some packages that are used by other parts of kubernetes codebase. 
+From now, we can point:
+* pkg/util/async (async bounder) which is used by pkg/controlplane
+* pkg/util/sysctl which is used by pkg/kubelet and kubemark
+
+### Move feature gate package dependency
+Because kube-proxy relies on defined Feature gates, it vendors `pkg/features`.
+
+This needs to be copied/moved the same way controller manager already does, moving this to
+`kubernetes/kube-proxy/pkg/features` and changing on the code, and putting all the feature
+gates referenced by kube-proxy there
+
+### Move pkg/proxy to staging
+The final step is to move pkg/proxy to `staging/src/k8s.io/kube-proxy/pkg` and let people willing
+to vendor kube-proxy to change from kubernetes/kubernetes to `kubernetes/kube-proxy` repo.
+
+### Risks and Mitigations
+"If a project vendors Kubernetes to import kubeproxy code, this will break them. 
+On the bright side, afterwards, these importers will have a much cleaner path to 
+include kubeproxy code. Before moving forward with this plan, we will identify 
+and communicate to these projects." - Shameless copied from 
```

Comment:
nit:
```suggestion
> If a project vendors Kubernetes to import kubeproxy code, this will break them. 
> On the bright side, afterwards, these importers will have a much cleaner path to 
> include kubeproxy code. Before moving forward with this plan, we will identify 
> and communicate to these projects.

&ndash;Shamelessly copied from 
```